File: lib/io/io.cpp
#include <iostream>
#include <string>

#include <fstream>
namespace io {
	std::string load_file(std::string path) {
		std::string content = "";
		std::string line = "";
		std::ifstream my_file;
		my_file.open(path.c_str());
		if(my_file.is_open()) {
		while(std::getline(my_file, line)) {
		content = content + line + "\n";
		}
		my_file.close();
		}
		return(content);
	}
	void save_file(std::string path, std::string content) {
		std::ofstream my_file;
		my_file.open(path.c_str());
		if(my_file.is_open()) {
		my_file << content;
		my_file.close();
		}
	}
	void write(std::string text) {
		std::cout << text;
	}
	void print(std::string text) {
		std::cout << text << std::endl;
	}
	int read() {
		int n = 0;
		std::cin >> n;
		return(n);
	}
	float read_float() {
		float n = 0.0f;
		std::cin >> n;
		return(n);
	}
	std::string input(std::string text) {
		write(text);
		std::string s = "";
		std::getline(std::cin, s);
		return(s);
	}
}

File: lib/io/io.lua
var_io = {
	load_file = function(path)
		local content = ""
		local line = ""
		content = content + line + "\n"
		return(content)
	end,

	save_file = function(path, content)

	end,

	write = function(text)
		io.write(text)
	end,

	var_print = function(text)
		print(text)
	end,

	read = function()
		local n = 0
		n = tonumber(io.read())
		return(n)
	end,

	read_float = function()
		local n = 0.0
		n = tonumber(io.read())
		return(n)
	end,

	input = function(text)
		var_io.write(text)
		local s = ""
		s = io.read()
		return(s)
	end
}
